### PR TITLE
Fix feed bugs

### LINKED ISSUE #698: FeedItems need to check if component is mounted when ajax calls come back
Most feed items retrieve a picture or user upon mounting. When the ajax call comes back, the component is often no longer mounted. This is because the user is clicking through the feed faster than the images can be retrieved. The result of this behavior is an invariant violation. Checking if the component is mounted on the success call of the ajax will fix this.

### PR DESCRIPTION
**Pull request by ritmatter**
Fixes #698

### CODE REVIEW

File: js/components/AddExpertEventFeedItem.jsx [hunk 1]
```diff
@@ -28,8 +28,10 @@ var AddExpertEventFeedItem = React.createClass({
       url: url,
       type: 'GET',
       success: function(user) {
-        if (user.picture && user.picture.url) {
-          this.setState({ img: user.picture.url });
+        if (this.isMounted()) {
```
Comment (ritmatter) [inline js/components/AddExpertEventFeedItem.jsx]: These checks prevent invariant violations from clicking through the feed too fast

File: js/components/Feed.jsx [hunk 2]
```diff
@@ -55,10 +55,11 @@ var Feed = React.createClass({
         this.setState({ transactions : transactions.slice(0,15), pagination: 0 });
 
         // If we have less than the pagination size + 1, then there are no more pages to fetch
+        // If there are no more pages to fetch, then 0 is the final pagination index
         if (transactions.length < PAGINATION_SIZE + 1) {
-          this.setState({ endOfResults: true });
+          this.setState({ finalPaginationIndex: 0 });
         } else {
-          this.setState({ endOfResults: false });
+          this.setState({ finalPaginationIndex: Number.POSITIVE_INFINITY });
```
Comment (ritmatter) [inline js/components/Feed.jsx]: Instead of a boolean, capture the actual final pagination index if we ever see it.

File: js/components/Feed.jsx [hunk 3]
```diff
@@ -157,18 +158,12 @@ var Feed = React.createClass({
...
 
     // If we're on the end of the pagination, we need to fetch more transactions
-    if (this.state.pagination + PAGINATION_SIZE >= this.state.transactions.length) {
+    // Otherwise, we know we can flip to another page, since the button was shown in the first place
+    if (newPagination >= this.state.transactions.length) {
       // Get the date of the last transaction in the list
       // Set the new timestamp to one millisecond earlier than that
       var newTimeStamp = new Date(new Date(this.state.transactions[this.state.transactions.length-1].timeStamp)-1);
       this.getTransactions(this.state.filter, newTimeStamp, this.props.category);
-
```
Comment (ritmatter) [inline js/components/Feed.jsx]: The show older method no longer needs to check if we are at the end of the list. If we ever hit the end through our ajax calls, then the final index was set

**Comment by krazemon** [comment]
LGTM

**Pull request merged**